mergify_engine/initial_configuration.py

```diff
@@ -0,0 +1,111 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+import yaml
+
+import github
+
+LOG = logging.getLogger(__name__)
+
+INITIAL_CONFIG_BRANCH = "mergify/initial-config"
+
+
+def sanitize_context(context):
+    # NOTE(sileht): Some CI report different for pr and branch
+    # We can't require branch testing on PR, so remove the detail
+    # * travis: remove /pr, /push
+    if context.endswith("/pr"):
+        context = context[:-3]
+    elif context.endswith("/push"):
+        context = context[:-5]
+    return context
+
+
+def create_pull_request(installation_token, repo):
+    pulls = list(repo.get_pulls(state="all"))[:5]
+    contexts = set()
+    for pull in pulls:
+        contexts.update(
+            sanitize_context(s.context)
+            for s in repo.get_commit(pull.head.sha).get_statuses()
+            if not s.context.startswith("mergify")
+        )
+
+    mergify_config = {
+        "rules": {
+            "default": {
+                "protection": {
+                    "required_status_checks": {
+                        "strict": True,
+                        "contexts": list(contexts),
+                    },
+                    "required_pull_request_reviews": {
+                        "required_approving_review_count": 1
+                    },
+                },
+                "merge_strategy": {
+                    "method": "rebase"
```

Comment:
Why is this different than the default config?

Reply:
I just copy/paste a configuration file, I have removed that.

Reply:
The problem is that most people will just click approve and not read the doc. So I don't think comfortable merging things that are not the default here.

Having mergify enabled with an (almost) empty file and then having them to poke around for more feature sounds like a safer approach

Reply:
so the file will looks like:

```
rules:
```